docs/sources/mimir/operators-guide/mimir-runbooks/_index.md

```diff
@@ -592,18 +592,37 @@ How to **investigate**:
+    - **How to detect**: Search compactor logs for `invalid result block` and `out-of-order chunks`.
+    - This is caused by a bug in the ingester - see [mimir#1537](https://github.com/grafana/mimir/issues/1537). Ingesters upload blocks where the MinT and MaxT of some chunks don't match the first and last samples in the chunk. When the faulty chunks' MinT and MaxT overlap with other chunks, the compactor merges the chunks. Because one chunk's MinT and MaxT are incorrect the merge may be performed incorrectly, leading to OoO samples.
+    - **How to mitigate**: Mark the faulty blocks to avoid compacting them in the future:
+      - Find all affected compaction groups in the compactor logs. You will find them as `invalid result block /data/compact/<compaction_group>/<result_block>`.
+      - For each failed compaction job
+        - Pick one result block (doesn't matter which)
+        - Find source blocks for the compaction job: search for `msg="compact blocks"` and a mention of the result block ID.
+        - Mark the source blocks for no compaction (in this example the object storage backend is GCS):
+          ```
+          ./tools/markblocks/markblocks -backend gcs -gcs.bucket-name <bucket> -mark no-compact -tenant <tenant-id> -details "Leading to out-of-order chunks when compacting with other blocks" <block-1> <block-2>...
+          ```
+
+Another possible cause of this alert is if compaction has never been running successfully, potentially for a long time. This can be because the alert either was not seen, misconfigured or not installed at all for some time.
+
+This can also result in long-term queries failing as the store-gateways fail to handle the much larger number of smaller blocks than expected.
+
+How to **investigate**:
+
+- Check the [Compactor Dashboard]({{< relref "../monitor-grafana-mimir/dashboards/compactor/index.md" >}})
+  - Compactor has fallen behind:
+    - **How to detect**:
+      - Check the `Last successful run per-compactor replica` panel - are there recent runs in the last 6-12 hours?
+      - Also check the `Average blocks / tenant` panel - what is the count? A tenant would have one 24h block per day plus smaller blocks to be compacted if all is well. Given that, <1200 blocks could be normal. Thousands and thousands of blocks is not normal.
+    - **What it means**: Compaction likely was failing for some reason in the past and now there is too much work to catch up at the current configuration and scaling level.
+    - **How to mitigate**: Reconfigure and modify the compactor settings and resources for more scalability:
+      - Ensure your compactors are at least sized according to the [Planning capacity]({{< relref "../run-production-environment/planning-capacity.md#compactor" >}}) page and you have the recommended number of replicas.
+      - Set `-compactor.split-groups` and `-compactor.split-and-merge-shards` to a multiple of 1 for every 25M active series in your deployment - so 100M series = value of `4`.
```

Comment:
@colega was considering making this 1 for every 8M AS. I also think 25M is too much.

Reply:
We took 25M because that's what saves the compactor from crashing when compacting, however those huge blocks are still suboptimal for the query path, especially when query sharding is enabled, and lower number is better.

I'd also state that this number should be an even number.

Reply:
Cool, I'll go with 8M then and edit this. Thanks!